railties/lib/rails/commands/server/server_command.rb

```diff
@@ -44,17 +71,24 @@ def set_environment
     end
 
     def start
+      missing_handler = false
+
       print_boot_information
       trap(:INT) { exit }
       create_tmp_directories
       setup_dev_caching
       log_to_stdout if options[:log_stdout]
 
       super
+    rescue LoadError
+      missing_handler = true
+      puts HandlerSuggestion.message(options[:server])
     ensure
-      # The '-h' option calls exit before @options is set.
-      # If we call 'options' with it unset, we get double help banners.
-      puts "Exiting" unless @options && options[:daemonize]
+      unless missing_handler
```

Comment:
I'm guessing the stopgap to inlining the HandlerSuggestion module in the command is this `unless`. But I don't fully understand why we need it? Isn't it fine for the server to say `Exiting` in the off case that a handler is missing?

Reply:
Actually, the stopgap is the `levenstein_distance` method being private and defined in the `Rails::Command::Behavior` module which carries command specific helpers. Decided to at least isolate it. I'd love to revisit the `levenshtein_distance` usage or swap it for the algo in `DidYouMean`, but I'd like to do it in a follow-up change.

The guard here is for my personal aesthetic reasons. 😅 Compare the two messages:

```
Could not load server "unicorn". Maybe you need the add it to the Gemfile?

  gem "unicorn"

Run `rails server --help` for more options.
Exiting
```

That `Exiting` line seems redundant to me, especially with `"Run rails server --help for more options."` as a final suggestion. Maybe the user mistyped an option and triggered this?

Can an option be to complicate the existing guard a bit and move the `missing_handler` check there?

```ruby
    # ...
    ensure
      # The '-h' option calls exit before @options is set.
      # If we call 'options' with it unset, we get double help banners.
      puts "Exiting" unless @options && options[:daemonize] || missing_handler
    end
```